src/VisualStudio/CSharp/Test/CodeModel/VisualStudioTestExportProvider.cs

```diff
@@ -15,7 +21,40 @@ static VisualStudioTestExportProvider()
         {
             Factory = ExportProviderCache.GetOrCreateExportProviderFactory(
                 TestExportProvider.EntireAssemblyCatalogWithCSharpAndVisualBasic.WithParts(
-                    ExportProviderCache.GetOrCreateAssemblyCatalog(typeof(CSharpCodeModelService).Assembly)));
+                    ExportProviderCache.GetOrCreateAssemblyCatalog(typeof(CSharpCodeModelService).Assembly)
+                        .WithPart(typeof(FakeVsServiceProvider))
```

Comment:
📝 The inclusion of this export resulted in `HACK_CSharpCreateServicesOnUIThread` having a complete set of imports available, and therefore became exposed as a `IWpfTextViewConnectionListener`. During tests, the use of `SVsServiceProvider` in the constructor of `AbstractSnippetInfoService` results in a crash that takes down the test process. I see two paths forward:

1. Implement `IServiceProvider.GetService` (challenging, and implementations of types like `SVsTextManager` might not be readily available in unit testing)
2. Exclude implementations of `HACK_AbstractCreateServicesOnUiThread` from the composition catalog (defers the crash in `AbstractSnippetInfoService` until such time as it's used)
3. Exclude implementations of `ISnippetInfoService` from the composition catalog (same, but indirect as this is the mechanism the "hack" uses to locate instances)

Reply:
💭 I lean towards option 3